src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudOneShotInstanceIT.java

```diff
@@ -0,0 +1,111 @@
+
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud;
+import com.google.jenkins.plugins.computeengine.client.ComputeClient;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.labels.LabelAtom;
+import hudson.tasks.Builder;
+import hudson.tasks.Shell;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+import java.util.logging.StreamHandler;
+import org.awaitility.Awaitility;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ComputeEngineCloudOneShotInstanceIT {
+  private static Logger log = Logger.getLogger(ComputeEngineCloudOneShotInstanceIT.class.getName());
+
+  @ClassRule public static Timeout timeout = new Timeout(5, TimeUnit.MINUTES);
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ByteArrayOutputStream logOutput = new ByteArrayOutputStream();
+  private static StreamHandler streamHandler = new StreamHandler(logOutput, new SimpleFormatter());
+  private static ComputeClient client;
+  private static Map<String, String> label = getLabel(ComputeEngineCloudOneShotInstanceIT.class);
+  private static FreeStyleBuild build;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    ComputeEngineCloud cloud = initCloud(jenkinsRule);
+    handleClassLogs(streamHandler, ComputeEngineCloud.class.getName());
+    client = initClient(jenkinsRule, label, log);
+    handleClassLogs(streamHandler, ComputeClient.class.getName());
+
+    cloud.addConfiguration(
+        instanceConfiguration(
+            DEB_JAVA_STARTUP_SCRIPT, NUM_EXECUTORS, LABEL, label, false, true, NULL_TEMPLATE));
+
+    jenkinsRule.jenkins.getNodesObject().setNodes(Collections.emptyList());
+
+    // Assert that there is 0 nodes
+    assertTrue(jenkinsRule.jenkins.getNodes().isEmpty());
+
+    FreeStyleProject project = jenkinsRule.createFreeStyleProject();
+    Builder step = new Shell("echo works");
+    project.getBuildersList().add(step);
+    project.setAssignedLabel(new LabelAtom(LABEL));
+
+    // Enqueue a build of the project, wait for it to complete, and assert success
+    build = jenkinsRule.buildAndAssertSuccess(project);
+  }
+
+  @AfterClass
+  public static void teardown() throws IOException {
+    ITUtil.teardown(streamHandler, logOutput, client, label, log);
+  }
+
+  @Test
+  public void testOneShotInstancesLogContainsExpectedOutput() throws Exception {
+    // Assert that the console log contains the output we expect
+    jenkinsRule.assertLogContains("works", build);
+  }
+
+  @Test
+  public void testOneShotInstanceNodeDeleted() {
```

Comment:
Suggest adding a test to verify the instance no longer exists in the compute client.